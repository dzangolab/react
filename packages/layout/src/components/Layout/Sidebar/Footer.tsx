import { LocaleSwitcher } from "@prefabs.tech/react-i18n";

import { Copyright, Version } from "../common";

type SidebarFooterProperties = {
  children?: React.ReactNode;
  noLocaleSwitcher?: boolean;
};

export const SidebarFooter = ({
  children,
  noLocaleSwitcher,
}: SidebarFooterProperties) => {
  const renderContent = () => {
    return (
      <>
        {!noLocaleSwitcher && <LocaleSwitcher />}
        <Copyright />
        <Version />
      </>
    );
  };

  return <div className="dz-sidebar-footer">{children || renderContent()}</div>;
};
