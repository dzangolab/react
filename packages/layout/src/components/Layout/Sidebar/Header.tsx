import { Logo } from "../common";
import { ToggleMenuMobile } from "../common/Toggle";

type SidebarHeaderProperties = {
  children?: React.ReactNode;
};

export const SidebarHeader = ({ children }: SidebarHeaderProperties) => {
  const renderContent = () => {
    return (
      <>
        <Logo />
        <ToggleMenuMobile />
      </>
    );
  };

  return <div className="sidebar-header">{children || renderContent()}</div>;
};
