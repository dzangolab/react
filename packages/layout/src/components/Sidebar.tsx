import React, { useState } from "react";

import "../css/components/sidebar.css";

interface Properties {
  children?: React.ReactNode;
  customStyle?: React.CSSProperties;
}

const Sidebar: React.FC<Properties> = ({ children, customStyle }) => {
  const [open, setOpen] = useState(true);

  return (
    <div
      className={open ? "sidebar-menu open" : "sidebar-menu"}
      style={customStyle}
    >
      <div className="toggle-button">
        <span onClick={() => setOpen(!open)}>&gt;</span>
      </div>
      {children}
    </div>
  );
};

export default Sidebar;
