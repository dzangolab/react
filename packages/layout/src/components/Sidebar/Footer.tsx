import useConfig from "@/hooks/useConfig";

import Copyright from "../Copyright";
import Version from "../Version";

const year = new Date().getFullYear();

export const SidebarFooter = () => {
  const appConfig = useConfig();

  return (
    <div className="sidebar-footer">
      <Copyright
        holder={appConfig?.copyright?.holder}
        url={appConfig?.copyright?.url}
        year={year}
      />
      <Version version={appConfig?.appVersion} />
    </div>
  );
};
