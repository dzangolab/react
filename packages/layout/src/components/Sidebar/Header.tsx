import { useMemo } from "react";

import useConfig from "@/hooks/useConfig";

import Logo from "../Logo";

type SidebarHeaderProperties = {
  expanded: boolean;
  header?: React.ReactNode;
  onToggle: () => void;
};

export const SidebarHeader = ({
  expanded,
  header,
  onToggle,
}: SidebarHeaderProperties) => {
  const { appName, layout: layoutConfig } = useConfig();

  const parseLogoAlt = useMemo(() => {
    if (!appName) {
      return;
    }

    return appName.replace("@", "").replace("/", " ");
  }, [appName]);

  return (
    <div className="sidebar-header">
      {header || (
        <Logo
          src={layoutConfig?.logo}
          route={layoutConfig?.logoRoute || layoutConfig?.homeRoute}
          alt={layoutConfig?.logoAlt || parseLogoAlt}
        />
      )}
      <span className="toggle" onClick={onToggle}>
        <i className={expanded ? "pi pi-times" : "pi pi-bars"}></i>
      </span>
    </div>
  );
};
