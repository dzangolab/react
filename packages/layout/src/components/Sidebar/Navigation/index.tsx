import { NavItem } from "./NavItem";

type NavigationMenuItem = {
  label: string;
  route: string;
  icon?: string;
};

type NavigationMenuGroup = {
  label: string;
  icon: string;
  submenu: NavigationMenuItem[];
};

export type NavigationMenu = Array<NavigationMenuItem | NavigationMenuGroup>;

export type SidebarNavigationProperties = {
  displayIcons?: boolean;
  navigationMenu: NavigationMenu;
  primaryNavigation?: boolean;
};

export const SidebarNavigation = ({
  displayIcons = true,
  navigationMenu,
  primaryNavigation = false,
}: SidebarNavigationProperties) => {
  const renderNavGroup = (navGroup: NavigationMenuGroup) => {
    return null;
  };

  const renderNavigation = (nav: NavigationMenuItem | NavigationMenuGroup) => {
    if ("submenu" in nav) {
      return renderNavGroup(nav);
    }

    return <NavItem navItem={nav} displayIcon={displayIcons} />;
  };

  if (primaryNavigation) {
    return (
      <nav className="sidebar-navigation">
        <ul>
          {navigationMenu.map((nav, index) => {
            return <li key={index}>{renderNavigation(nav)}</li>;
          })}
        </ul>
      </nav>
    );
  }

  return (
    <div className="sidebar-navigation">
      <ul>
        {navigationMenu.map((nav, index) => {
          return <li key={index}>{renderNavigation(nav)}</li>;
        })}
      </ul>
    </div>
  );
};
