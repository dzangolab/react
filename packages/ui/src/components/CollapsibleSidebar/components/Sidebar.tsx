import React from "react";

import "../index.css";

interface Properties {
  children?: React.ReactNode;
  customStyle?: React.CSSProperties;
  open: boolean;
}

const Sidebar: React.FC<Properties> = ({ children, customStyle, open }) => {
  return (
    <div
      className={open ? "sidebarMenu openSidebarMenu" : "sidebarMenu"}
      style={customStyle}
    >
      {children}
    </div>
  );
};

export default Sidebar;
