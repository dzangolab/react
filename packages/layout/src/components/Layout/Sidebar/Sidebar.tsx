import { SidebarFooter } from "./Footer";
import { SidebarHeader } from "./Header";
import { NavigationMenu } from "../common";
import { UserMenu } from "../common/UserMenu";

import type { NavMenuItemType, NavMenuType, UserMenuModeType } from "../types";

type SidebarProperties = {
  children?: React.ReactNode;
  collapsible?: boolean;
  displayNavIcons?: boolean;
  navigationMenu?: NavMenuType;
  noFooter?: boolean;
  noHeader?: boolean;
  noLocaleSwitcher?: boolean;
  userMenu?: NavMenuItemType;
  userMenuMode?: "popup" | "collapsible" | "collapsible-reverse";
  trigger?: React.ReactNode;
};

export const Sidebar = ({
  children,
  collapsible = true,
  displayNavIcons = false,
  navigationMenu,
  noFooter = false,
  noHeader = false,
  noLocaleSwitcher = false,
  userMenu,
  userMenuMode,
  trigger,
}: SidebarProperties) => {
  const renderContent = () => {
    return (
      <>
        {!noHeader && <SidebarHeader />}
        <NavigationMenu
          displayIcons={displayNavIcons}
          navigationMenu={navigationMenu || []}
        />
        {userMenu && (
          <UserMenu
            menu={userMenu}
            trigger={trigger}
            userMenuMode={userMenuMode}
          />
        )}
        {!noFooter && <SidebarFooter noLocaleSwitcher={noLocaleSwitcher} />}
      </>
    );
  };

  return (
    <div className={`dz-sidebar ${!collapsible ? "fixed" : ""}`.trim()}>
      {children || renderContent()}
    </div>
  );
};
