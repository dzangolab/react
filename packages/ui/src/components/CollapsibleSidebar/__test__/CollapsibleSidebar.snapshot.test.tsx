import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import Sidebar from "..";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot sidebar closed", () => {
  const component = create(
    <Sidebar open={false}>
      <ul>
        <li>Home</li>
        <li>About</li>
      </ul>
    </Sidebar>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test("Component matches snapshot sidebar open", () => {
  const component = create(
    <Sidebar open={true}>
      <ul>
        <li>Home</li>
        <li>About</li>
      </ul>
    </Sidebar>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
