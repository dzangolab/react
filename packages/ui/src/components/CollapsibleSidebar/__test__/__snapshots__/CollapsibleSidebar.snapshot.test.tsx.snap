// Vitest Snapshot v1

exports[`Component matches snapshot sidebar closed 1`] = `
<div
  className="sidebarMenu"
/>
`;

exports[`Component matches snapshot sidebar open 1`] = `
<div
  className="sidebarMenu openSidebarMenu"
>
  <ul>
    <li>
      Home
    </li>
    <li>
      About
    </li>
  </ul>
</div>
`;
